### PR TITLE
mcrockett/Add more config options, Ruby 3 compatible

### PR DESCRIPTION
**Pull request by mmcrockett**
- Added ability to configure endpoint and authorization overrides
- Now Ruby 3.x compatible

### CODE REVIEW

**Review by cmrr95 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+# [5.4.3] - 2024-03-06
```
Comment (vechema) [inline CHANGELOG.md:8]:
```suggestion
# [5.5.0] - 2024-03-06
```
It's not a bug fix

File: CHANGELOG.md [hunk 2]
```diff
@@ -599,6 +607,7 @@ Added the ability to hit professional claim submission API. For more details, se
 * Authentication
 * Configuration
 
+[5.4.3]: https://github.com/WeInfuse/change_health/compare/v5.4.2...v5.4.3
```
Comment (vechema) [inline CHANGELOG.md:610]:
```suggestion
[5.5.0]: https://github.com/WeInfuse/change_health/compare/v5.4.2...v5.5.0
```

File: lib/change_health/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module ChangeHealth
-  VERSION = '5.4.2'.freeze
+  VERSION = '5.4.3'.freeze
```
Comment (vechema) [inline lib/change_health/version.rb:2]:
```suggestion
  VERSION = '5.5.0'.freeze
```

**Review by vechema — APPROVED** [review]
Cool, then I'll need to add the endpoint flexibility for submission & reports!

lgtm

**Comment by mmcrockett** [comment]
Good call! Didn't really think about it. Thought about majorness but not minor/bugfix.

**Pull request merged**